assets/bigquery/encounter-events.sql.j2

```diff
@@ -1,93 +1,227 @@
 #standardSQL
+--
+-- Creates a port-id based in the iso3 and the label of the port
+--
+CREATE TEMPORARY FUNCTION
+  generate_port_id (iso3 STRING,
+    label STRING) AS ( LOWER(CONCAT(iso3,"-",REGEXP_REPLACE(NORMALIZE(label),' ',''))) );
 WITH
-  #
-  # Good encounter events
-  #
+  --
+  -- Declare the source tables we are going to use through the entire query
+  --
+  source_encounters AS (
+  SELECT
+    *
+  FROM
+    `{{ source }}`),
+  source_vessel_info AS (
+  SELECT
+    *
+  FROM
+    `{{ vessel_info }}`),
+  source_spatial_measures AS (
+  SELECT
+    *
+  FROM
+    `{{ spatial_measures }}`),
+  source_country_codes AS (
+  SELECT
+    *
+  FROM
+    `{{ country_codes }}` ),
+  source_voyages AS (
+  SELECT
+    *
+  FROM
+    `{{ voyages }}`),
+  source_anchorages AS (
+  SELECT
+    *
+  FROM
+    `{{ named_anchorages }}`),
+  --
+  -- We need to flatten voyages so that we have a record for each vessel id
+  --
+  flattened_voyages AS (
+  SELECT
+    trip_id,
+    trip_start,
+    trip_end,
+    trip_start_anchorage_id,
+    trip_end_anchorage_id,
+    vessel_id
+  FROM
+    source_voyages
+  CROSS JOIN
+    UNNEST(vessel_ids) AS vessel_id ),
+  --
+  -- Good encounter events
+  --
   encounters AS (
   SELECT
     *
   FROM
-    `{{ source }}`
+    source_encounters
   WHERE
     TIMESTAMP_DIFF(end_time, start_time, SECOND) > (3600 * 2)
-    AND median_speed_knots < 2 ),
-
-  #
-  # Duplicate encounters so that we have an event for each vessel
-  #
+    AND median_speed_knots < 2),
```

Comment:
@andres-arana let's parametrize this speed, in the encounters tileset we are using 2.4 knots.